### PR TITLE
allow bringing an existing instance role & profile

### PR DESCRIPTION
**Pull request by anniehedgpeth**
## Background

Closes Issue #284 
Jira [TFE-344](https://hashicorp.atlassian.net/browse/TFE-344)

This branch allows for the user to bring their own existing IAM instance profile and/or an IAM role used for the instance profile. The new variables that enable this behavior are:
- `existing_iam_instance_profile_name`
- `existing_iam_instance_role_name`

## How Has This Been Tested

I tested this locally by passing the new variables into my test module. I looked in the AWS portal to confirm that the Autoscaling Launch Configuration's IAM instance profile was the same one that I passed in.

I will also ensure that a passing test is run from the comments to ensure this doesn't break existing functionality.

[TFE-344]: https://hashicorp.atlassian.net/browse/TFE-344?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by nathanl79 — APPROVED** [review]

File: modules/service_accounts/data.tf
Comment (theoskolnik) [inline modules/service_accounts/data.tf:12]: Not a blocker, would help to modify these resources with the term "existing" so that later on this could read like: 

`iam_instance_role    = try(data.aws_iam_role.existing_instance_role[0], aws_iam_role.instance_role[0])`

**Review by theoskolnik — APPROVED** [review]
Nothing blocking, just readability, nice job!

File: modules/service_accounts/data.tf
Comment (anniehedgpeth) [reply modules/service_accounts/data.tf:12]: Ah, yeah, I like that. Thanks! I'll add it on Monday when I'm back from PTO. (Headed out now.)

Comment (anniehedgpeth) [reply modules/service_accounts/data.tf]: I had time after all, but that means I need another approval. :)

**Review by sudomateo — APPROVED** [review]
I'm of the opinion that we should drop the `existing_` prefix on the variables since the presence of a variable value indicates that it is existing. It's not something that I would block merge on though so I'll leave the decision up to you @anniehedgpeth.

**Comment by anniehedgpeth** [comment]
I hear you, but there are some places that a "*_name" variable is for the preferred name of a new resource, so I think I'll keep it like this for now.

**Pull request merged**
